src/pybamm/meshes/meshes.py

```diff
@@ -169,7 +171,12 @@ def combine_submeshes(self, *submeshnames):
         # Check that the final edge of each submesh is the same as the first edge of the
         # next submesh
         for i in range(len(submeshnames) - 1):
-            if self[submeshnames[i]].edges[-1] != self[submeshnames[i + 1]].edges[0]:
+            if self[submeshnames[i]].edges[-1] == self[submeshnames[i + 1]].edges[0]:
+                pass
+            elif hasattr(self[submeshnames[i + 1]], "min"):
```

Comment:
this isn't totally clear to me - what's this catching?

Reply:
I tried to make it a bit more clear -- basically we want sub meshes to pass iff they are aligned or if either is a symbolic sub mesh (we cannot check symbolic submeshes at compile-time, so we have to give them the benefit of the doubt and always let them pass)